tests/components/azure_data_explorer/test_init.py

```diff
@@ -0,0 +1,280 @@
+"""Test the init functions for AEH."""
+from datetime import timedelta
+import logging
+from unittest.mock import patch
+
+from azure.kusto.data.exceptions import KustoAuthenticationError, KustoServiceError
+import pytest
+
+from homeassistant.components import azure_data_explorer
+from homeassistant.components.azure_data_explorer.const import (
+    CONF_SEND_INTERVAL,
+    DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import STATE_ON
+from homeassistant.setup import async_setup_component
+from homeassistant.util.dt import utcnow
+
+from .conftest import FilterTest
+from .const import AZURE_DATA_EXPLORER_PATH, BASE_CONFIG_FULL, BASIC_OPTIONS
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    ("sideeffect"),
+    [None, KustoServiceError("test"), KustoAuthenticationError("test", Exception)],
+    ids=["No_error", "KustoServiceError", "KustoAuthenticationError"],
+)
+async def test_put_event_on_queue_with_managed_client(
+    hass,
+    entry_managed,
+    mock_azure_data_explorer_ManagedStreamingIngestClient_ingest_data,
+    sideeffect,
+) -> None:
+    """Test listening to events from Hass. and writing to ADX with managed client."""
+
+    mock_azure_data_explorer_ManagedStreamingIngestClient_ingest_data.side_effect = (
```

Comment:
What is this test testing and what happens with the different side_effects?

Reply:
And will these errors be silently ignored? If not, we should assert for it. We can assert for log entries as example

Reply:
Assert for log entries would be valid. I cant find any documentation on how to assert for that, can you provide an example?

Reply:
There are many example in the current tests. Search for `caplog: pytest.LogCaptureFixture`